ui-core: fix focus style on button

turnerian2004 opened the pull request:
closes #507 

@SharglutDev Do you have a figma file you can share? What should be the amount of space between the button & the outline? What is the width of the outline border? What are the border colors?

I took a look at the [outline-offset](https://developer.mozilla.org/en-US/docs/Web/CSS/outline-offset). The css property seems to not have a unique border radius property. Instead, outline-offset seems to inherent the border radius of the component's border. In the screen shot you shared, the offset has a square border whereas the button has a rounded border. What are your thoughts?  



https://github.com/user-attachments/assets/f043adb8-558d-4c2c-89f8-fba7c3f1da37

SharglutDev commented:
Hi @turnerian2004, thanks for taking a look in this issue. 

We work with sketch on osrd, I'll check with our UX @thibautsailly if he's ok to open some additional access for external contributors and will get back to you asap.

turnerian2004 commented:
> Hi @turnerian2004, thanks for taking a look in this issue (I edited it with the link of the mockup for buttons).
> 
> We work with sketch on osrd, I'll check with our UX @thibautsailly if he's ok to open some additional access for external contributors and will get back to you asap.

Thank you very much for the info @SharglutDev. The link doesn't currently work, but I'll check back later this week if the link works :)

SharglutDev commented:
> > Hi @turnerian2004, thanks for taking a look in this issue (I edited it with the link of the mockup for buttons).
> > We work with sketch on osrd, I'll check with our UX @thibautsailly if he's ok to open some additional access for external contributors and will get back to you asap.
> 
> Thank you very much for the info @SharglutDev. The link doesn't currently work, but I'll check back later this week if the link works :)

Yep, it's normal. The subject is a little touchy, we have some private stuff on sketch so @thibautsailly is currently checking if he can setup a special sketch account for public stuff only. We'll get back to you asap.

turnerian2004 commented:
> > > Hi @turnerian2004, thanks for taking a look in this issue (I edited it with the link of the mockup for buttons).
> > > We work with sketch on osrd, I'll check with our UX @thibautsailly if he's ok to open some additional access for external contributors and will get back to you asap.
> > 
> > 
> > Thank you very much for the info @SharglutDev. The link doesn't currently work, but I'll check back later this week if the link works :)
> 
> Yep, it's normal. The subject is a little touchy, we have some private stuff on sketch so @thibautsailly is currently checking if he can setup a special sketch account for public stuff only. We'll get back to you asap. (might take a few days)

@SharglutDev I don't have access to the file. Do you want to take over the pr? I do not object from my side :)

thibautsailly commented:
Hello, sorry for being late in my answer.

About sharing design files with external contributors: I'm guessing you would have liked to have access to a developper handoff feature where you can inspect properties, rather than having a static image with written specs. This still needs to be organised.

About this specific issue: the only design decision I made on this topic is to have a 2px border in primary50 colour, placed 2px from the button to ensure there would be enough contrast between the rest state and the focused one. If a browser natively provides enough visual difference between these two states, it can remain as is.

If having a border radius is easier, I don't see it as an issue – I made a square border because I thought it would be easier to implement.

Here is a closer view to the mockup:
<img width="566" alt="image" src="https://github.com/user-attachments/assets/d7674d8c-181d-46fa-b349-9d43b1804042" />

Commit:
ui-core: fix focus style on button

Signed-off-by: Ian Turner <turnerian2004@gmail.com>

SharglutDev commented on ui-core/src/styles/button.css:
```diff
@@ -28,6 +28,10 @@
   }
   &.primary {
+    &:focus {
+      outline: 0.125rem solid #256afa;
```
This file is a little outdate but a few months ago we decided to use pixels for all css except `font-size`, we can start to use it right now in this file (the rest can be done a in different PR).
Same for colors, we should use this syntax.

You can do the same for the rest of the css you changed in this PR :)
```suggestion
      outline: 2px solid theme('colors.primary.50');
```

turnerian2004 replied:
Thanks for the feedback @SharglutDev .... I will work on this later this week when I have time :)

SharglutDev commented on ui-core/src/styles/button.css:
```diff
@@ -61,6 +65,10 @@
   }
   &.cancel {
+    &:focus {
+      outline: 0.125rem solid #797671;
```
It wasn't specified but the outline can keep the same color all the time (primary 50) even though it isn't a primary one

turnerian2004 commented:
@SharglutDev Thanks for the feedback on the pr. I think I addressed your concerns: 

1. 2 px border - instead of using rem
2. always use colors.primary.50 for the border colour
3. 2px offset

I saw this repo seems to no longer be [active](https://github.com/OpenRailAssociation/osrd-ui/commit/ff8b8cc90e0c50119b79018c534341105feb9b8d). Should I open this pr [here](https://github.com/OpenRailAssociation/osrd/pulls) instead?

Thanks again for taking the time to walk me through this issue. Greatly appreciated.

SharglutDev commented:
Sorry @turnerian2004, didn't have time to review your changes today.

> I saw this repo seems to no longer be [active](https://github.com/OpenRailAssociation/osrd-ui/commit/ff8b8cc90e0c50119b79018c534341105feb9b8d). Should I open this pr [here](https://github.com/OpenRailAssociation/osrd/pulls) instead?

Your totally right, we integrated osrd-ui inside osrd direclty.
To migrate your PR, you can follow these steps : 
```
One can migrate a PR from osrd-ui to osrd like so:

Create a new branch in osrd
Run the command below to apply the commits locally
Push to GitHub
Open a new PR with a link to the old osrd-ui PR in the description
Assign reviewers

pr=976 (replace by your PR number)
curl -L "https://github.com/OpenRailAssociation/osrd-ui/pull/$pr.patch" | git am --directory=front/ui
```

emersion commented:
Ah, sorry, I forgot to post a notice. More details are available here: https://github.com/OpenRailAssociation/osrd/pull/10711#migrate-pr
